src/clang_parser.cpp

```diff
@@ -234,6 +260,9 @@ void ClangParser::parse(ast::Program *program, StructMap &structs)
           auto ptypestr = get_clang_string(clang_getTypeSpelling(ptype));
           auto ptypesize = clang_Type_getSizeOf(ptype);
 
+          if(clang_Cursor_isAnonymous(parent))
+            offset = get_indirect_field_offset(c, struct_name);
```

Comment:
perhaps this could be simplified by having a single function to calculate offsets, as I've done above for struct names?